### DISCUSSION TITLE
Notify worker thread

### DISCUSSION THREAD

**Discussion by jim-king-2000** [libuv/libuv#4379] [Q&A] [answered]
Here is an example of notifying main thread from worker thread: <https://docs.libuv.org/en/v1.x/guide/threads.html#inter-thread-communication>.
Is there any example of notifying worker thread from main thread?

**Comment by bnoordhuis** [comment]
Not by means of a mechanism similar to `uv_async_send()` because the worker thread doesn't have an event loop but you can use something like a condition variable and a mutex (`uv_cond_t` and `uv_mutex_t`).

Whether that's a good idea is up for debate. The thread pool is fixed size and sleeping in `uv_cond_wait()` means you're hogging one. Of course it's fine if you create the thread yourself with `uv_thread_create()`.

**Reply by jim-king-2000** [reply]
Thanks for the quick reply. What is the best practice for notifying child thread in libuv? Is it acceptable to add an event loop for the work thread (like <https://github.com/kristrev/libuv-multiple-loops/blob/master/libuv-test.c>)?

**Reply by bnoordhuis** [reply, marked as answer]
That could work with the caveat that the event loop should be created on the worker thread, not the main thread and then passed to the worker thread.
